tests/test_model_upload.py

```diff
@@ -142,3 +143,41 @@ def test_model_upload_with_directory_structure(self) -> None:
             # TODO: Add assertions on CreateModelInstanceRequest.Directories and
             # CreateModelInstanceRequest.Files to verify the expected structure
             # is sent.
+
+    def test_model_upload_with_ignore_patterns(self) -> None:
+        with TemporaryDirectory() as tmp_dir:
+            tmp_dir_p = Path(tmp_dir)
+            # files to upload
+            (tmp_dir_p / "a" / "b").mkdir(parents=True)
+            (tmp_dir_p / "weights.txt").touch()
+            (tmp_dir_p / "a" / "a.txt").touch()
+            (tmp_dir_p / "a" / "b" / "b.txt").touch()
+            (tmp_dir_p / "a" / "b" / ".bb").touch()
+            expected_files = {
+                "weights.txt",
+                "a.txt",
+                "b.txt",
+                ".bb",
+            }
+
+            # files to ignore
+            (tmp_dir_p / ".git").mkdir(parents=True)
+            (tmp_dir_p / ".git" / "file").write_text("hidden git file")
```

Comment:
We should also ignore any `.git` folder in subdirectories.

Reply:
Done. Added more tests for .git in subdirs.